packages/frontend-web/src/app/components/AssignTagsForm/AssignTagsForm.tsx

```diff
@@ -101,31 +158,42 @@ export class AssignTagsForm extends React.Component<IAssignTagsFormProps, IAssig
     }
   }
 
-  render() {
+  @autobind
+  submit() {
+    const {tagsPreselected, selectedTagIds} = this.state;
+    if (selectedTagIds.size === 0) {
+      return;
+    }
+    const rejectedTagIds = tagsPreselected ? tagsPreselected.subtract(selectedTagIds) : Set<ModelId>();
+    this.props.submit(this.props.comment.id, selectedTagIds, rejectedTagIds);
+  }
 
+  render() {
     const { tags } = this.props;
     const { selectedTagIds } = this.state;
 
+    const enabled =  selectedTagIds.size > 0;
     return (
-      <div {...css(STYLES.container)}>
-        <ul {...css(STYLES.tagsList)}>
-          {tags && tags.map((t) => (
-            <li key={`tag${t.id}`} {...css(STYLES.listItem)}>
-              <CheckboxRow
-                label={t.label}
-                isSelected={selectedTagIds && selectedTagIds.includes(t.id)}
-                onChange={partial(this.onTagButtonClick, t.id)}
-              />
-            </li>
-          ))}
-        </ul>
-        <Button
-          buttonStyles={STYLES.button}
-          label="Reject Comment"
-          disabled={selectedTagIds && selectedTagIds.size === 0}
-          onClick={partial(maybeCallback(this.props.onSubmit), selectedTagIds)}
-        />
-      </div>
+      <ClickAwayListener onClickAway={this.props.clearPopups}>
+        <div {...css(SCRIM_STYLE.popupMenu, {padding: '20px 60px'})}>
+          <DialogTitle id="article-controls">Reason for rejection</DialogTitle>
+          <ul {...css(STYLES.tagsList)}>
+            {tags && tags.map((t) => (
+              <li key={`tag${t.id}`} {...css(STYLES.listItem)}>
```

Comment:
I think you probably want `key="{\`tag${t.id}\`}"` to make sure that any strange characters in a tag don't break here.

Reply:
Hi Lucas.  I don't understand this review comment.  t.id should be a number or simple string.  So key should be set to "tag1" or something similar...

Reply:
I think if tag is guaranteed to not have spaces, it's fine. But if it does, then it will break.